docker/Dockerfile.build

```diff
@@ -0,0 +1,64 @@
+FROM ubuntu:18.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+WORKDIR /usr/src
+
+RUN set -ex \
+	&& apt-get update \
+	&& apt-get -y dist-upgrade \
+	&& apt-get install -qq --no-install-recommends \
+	git ca-certificates \
+	&& rm -rf /var/lib/apt/lists/*
+
+RUN git clone https://github.com/OrchidTechnologies/orchid.git
+
+WORKDIR orchid
+
+RUN git config --global url."git://git.savannah.gnu.org/".insteadOf "https://git.savannah.gnu.org/git/" \
+	&& git config --global url."git://git.savannah.nongnu.org/".insteadOf "https://git.savannah.nongnu.org/git/" \
+	&& git config --global advice.detachedHead false
+
+RUN git submodule update --init --recursive --jobs 3
+
+RUN set -ex \
+	&& apt-get update \
+	&& apt-get install -qq --no-install-recommends \
+	build-essential \
+	clang-8 \
+	clang++-8 \
+	clang-tidy-8 \
+	libc++-8-dev \
+	libc++abi-8-dev \
+	bc \
+	libboost-all-dev \
+	bison \
+	flex \
+	gettext \
+    gperf \
+    groff \
+    ninja-build \
+    python3-pip \
+    python3-setuptools \
+    tcl \
+	&& rm -rf /var/lib/apt/lists/*
+
+RUN update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-8 100
+RUN update-alternatives --install /usr/bin/clang clang /usr/bin/clang-8 100
+RUN update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-8 100
+
+RUN set -ex \
+	&& perl -pi -e 's/\$\(llvm\)\/bin\/clang-tidy/\/usr\/bin\/clang-tidy/g' env/output.mk \
+	&& perl -pi -e 's/,--icf=all//g' env/target-lnx.mk
```

Comment:
I'd love for @saurik to weigh in on this bit. The build system is very configurable, and I wouldn't be surprised if there isn't already a nice variable that can be set in `local.mk`.

Reply:
This is a set of two workarounds, one for a bug in the build scripts that I had already fixed and one that I currently consider incorrect: people should not be doing either of these.

Reply:
(Somehow when I saw this comment on my phone I was seeing it on two different lines of code that were also unnecessary and could instead be replaced by setting suffix=.)

Reply:
Removed.  These were needed back when I originally did this, but not any more.